bin/git-feature

```diff
@@ -29,6 +35,15 @@ do
       remote="origin"
     fi
     ;;
+  -s|--separator )
+    if [[ -n $2 ]]
+    then
+      branch_separator=$2
+      shift
+    else
+      argv+=("$1") # treat tail '-s' as <name>
```

Comment:
I think we can remove this. Otherwise, adding new option will be cumbersome.

Reply:
What should we do if we get `git feature -s` or `git feature xyz -s`?

I had initially treated a missing second parameter as an error, but then I saw how `-a` was handled so I made it consistent.

My personal opinion is that the handling should be consistent, but that it should be as an error *unless* we also check for `--` to end option parsing.

Something like this (this is untested, but based on a pattern that I have written fairly frequently):

```diff
diff --git bin/git-feature bin/git-feature
index f33015c94902..0751ce32761a 100755
--- bin/git-feature
+++ bin/git-feature
@@ -18,16 +18,17 @@ while test $# != 0
 do
   case $1 in
   -a|--alias )
-    if [[ -n $2 ]]
+    if [[ -n $2 ]] && [[ $2 != -- ]]
     then
       shift # shift -a|-alias
       branch_prefix=$1
     else
-      argv+=("$1") # treat tail '-a' as <name>
+      echo >&2 "option $1 requires a value"
+      exit 1
     fi
     ;;
   -r|--remote )
-    if [[ -n $2 ]]
+    if [[ -n $2 ]] && [[ $2 != -- ]]
     then
       remote=$2
       shift
@@ -36,12 +37,13 @@ do
     fi
     ;;
   -s|--separator )
-    if [[ -n $2 ]]
+    if [[ -n $2 ]] && [[ $2 != -- ]]
     then
       branch_separator=$2
       shift
     else
-      argv+=("$1") # treat tail '-s' as <name>
+      echo >&2 "option $1 requires a value"
+      exit 1
     fi
     ;;
   --squash )
@@ -51,6 +53,12 @@ do
     start_point=$2
     shift
     ;;
+  -- )
+    # terminate argument parsing
+    shift
+    argv+=("$@")
+    break
+    ;;
   * )
     argv+=("$1")
     ;;
```

Reply:
Let's try the diff you provide.

Reply:
I’ve added this as a second commit. When I was writing up the documentation to capture this properly, I came across some test cases which required a few more changes than originally provided.

If the further changes are accepted, I can squash this into one commit before merge with a cleaner commit message.

There are other changes that I would *personally* make (using the result of `shfmt`, using `(($#))` instead of `test $# != 0`, always using `[[…]]` instead of sometimes using `[…]`, a few other things), but I have avoided doing those in this PR because it would obfuscate the logic changes.